src/sys/pthread.rs

```diff
@@ -43,17 +43,24 @@ pub fn pthread_kill<T: Into<Option<Signal>>>(thread: Pthread, signal: T) -> Resu
 ///
 /// On linux the name cannot exceed 16 length including null terminator
 #[cfg(all(target_os = "linux", not(target_env = "musl")))]
-pub fn pthread_getname_np(thread: Pthread) -> Result<String> {
-    let mut name = [0u8; 16];
+pub fn pthread_getname_np(thread: Pthread) -> Result<CString> {
+    let mut name: Vec<u8> = Vec::with_capacity(16);
+    unsafe { name.set_len(16); }
     unsafe { libc::pthread_getname_np(thread, name.as_mut_ptr() as _, name.len()) };
-    let cname = unsafe { CStr::from_ptr(name.as_ptr() as _) };
-    Ok(cname.to_owned().to_string_lossy().to_string())
+    let zi = name.iter().position(|x| *x == b'\0').unwrap();
```

Comment:
This three lines can be simplified.  If you preinitialize the buffer to zero, then all you'll have to do is `CString::new(name)`.